src/main/java/com/iyzipay/HashGenerator.java

```diff
@@ -1,14 +1,30 @@
 package com.iyzipay;
 
-import org.apache.commons.codec.binary.Base64;
-import org.apache.commons.codec.digest.DigestUtils;
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import javax.xml.bind.DatatypeConverter;
 
 public final class HashGenerator {
 
     private HashGenerator() {
     }
 
     public static String generateHash(String apiKey, String secretKey, String randomString, Object request) {
-        return Base64.encodeBase64String(DigestUtils.sha1(apiKey + randomString + secretKey + request));
+        String input = apiKey + randomString + secretKey + request;
+        StringBuilder sb = new StringBuilder();
+        byte[] result = null;
+        try {
+            MessageDigest crypt = MessageDigest.getInstance("SHA1");
+            result = crypt.digest(input.getBytes());
+            for (int i = 0; i < result.length; i++) {
+                sb.append(Integer.toString((result[i] & 0xff) + 0x100, 16).substring(1));
+            }
+        } catch (NoSuchAlgorithmException ex) {
+            Logger.getLogger(HashGenerator.class.getName()).log(Level.SEVERE, null, ex);
```

Comment:
Please throw iyzipay HttpClientException. Do not log anything. Client will handle this.